homeassistant/helpers/entity.py

```diff
@@ -1418,6 +1418,11 @@ async def async_internal_added_to_hass(self) -> None:
 
         entity_sources(self.hass)[self.entity_id] = entity_info
 
+        if "_extra" in self.__combined_unrecorded_attributes:
+            new = {*self.__combined_unrecorded_attributes}
+            new.remove("_extra")
+            new.union(self._attr_extra_state_attributes.keys())
```

Comment:
```suggestion
            new.union(self.extra_state_attributes.keys())
```

Doesn't this assume the entity is using `_attr` shorthand?

Reply:
Yes. It was just a quick and dirty to have a discussion around but I didn't come back to it yet. It's not in a state that it would be any good.